src/basegraph.cc

```diff
@@ -257,6 +257,13 @@ namespace dd
           outputdim.push_back(_graph[inputs[0]].dim[i]);
         outputdim.push_back(_graph[producer].num_output);
       }
+    else if (_graph[producer].type == "Tile")
+      {
+        // BIG FAT WARNING : this is a hack for tile in autoencoders only
```

Comment:
Maybe we may condition on an `_autoencoder` boolean ?

Reply:
my first aim was to implement generic Tile layer, independently from autoencoders, but it needs a few changes that I will do later (I have a ticket :)  , so I would prefer to stick with current version here, and clean up things when I will allow different sizes for outputs of a layer ( it is not complicated, but needs to be done carefully)

Reply:
OK